### PR TITLE
Flagging prep eh

### PR DESCRIPTION
**Pull request by ehinman**
Mostly changes to flagging functions so they always return a flag column that is populated with something.

### CODE REVIEW

File: R/CensoredDataSuite.R [hunk 1]
```diff
@@ -213,11 +213,11 @@ summarizeCensoredData <- function(.data, spec_cols = c("TADA.CharacteristicName"
...
-    Percent_Censored<50&Censoring_Levels>1 ~ as.character("Kaplan-Meier"),
-    Percent_Censored<50&Sample_Count>=50 ~ as.character("Robust Regression Order Statistics"),
-    Sample_Count<50 ~ as.character("Robust Regression Order Statistics"),
-    Percent_Censored>=50&Sample_Count<50 ~ as.character("Maximum Likelihood Estimation")
-  ))
+    Percent_Censored>80 ~ as.character("Percent censored too high for estimation methods"), # greater than 80, cannot estimate
+    Percent_Censored<50&Censoring_Levels>1 ~ as.character("Kaplan-Meier"), # less than 50% censored, and multiple censoring levels (no minimum n)
+    Percent_Censored<50 ~ as.character("Robust Regression Order Statistics"), # less than 50% censored and one censoring level (no minimum n?)
+    Sample_Count>=50 ~ as.character("Maximum Likelihood Estimation"), # 50%-80% censored, 50 or more samples
+    Sample_Count<50 ~ as.character("Robust Regression Order Statistics"), # 50%-80% censored, less than 50 samples
```
Comment (ehinman) [inline R/CensoredDataSuite.R:220]: I had this decision tree a little off, so fixed.

File: R/ResultFlagsDependent.R [hunk 2]
```diff
@@ -244,17 +244,17 @@ InvalidSpeciation <- function(.data, clean = c("invalid_only", "nonstandardized_
   #                   .after = "MethodSpecificationName"
   #   )
   
-  # if all rows are "Valid", return input unchanged
+  # if all rows are "Valid", return input with flag column
   if (any(c("Nonstandardized", "Invalid") %in%
           unique(check.data$TADA.MethodSpeciation.Flag)) == FALSE) {
-    print("All data is valid, therefore the function cannot be applied.")
-    .data = OrderTADACols(.data)
+    print("All characteristic/method speciation combinations are valid in your dataframe. Returning input dataframe with TADA.MethodSpeciation.Flag column.")
```
Comment (cristinamullin) [inline R/ResultFlagsDependent.R:250]: Suggestion: Returning input dataframe with added TADA.MethodSpeciation.Flag column for tracking.

File: R/ResultFlagsDependent.R [hunk 3]
```diff
@@ -94,22 +94,22 @@ InvalidFraction <- function(.data, clean = TRUE, errorsonly = FALSE) {
   if (any(c("Nonstandardized", "Invalid") %in%
           unique(check.data$TADA.SampleFraction.Flag)) == FALSE) {
     if (errorsonly == FALSE) {
-      print("No changes were made, because we did not find any invalid fraction/characteristic combinations in your dataframe")
-      .data = OrderTADACols(.data)
-      return(.data)
+      print("All characteristic/fraction combinations are valid in your dataframe. Returning input dataframe with TADA.SampleFraction.Flag column.")
```
Comment (cristinamullin) [inline R/ResultFlagsDependent.R:97]: Suggestion: Returning input dataframe with added TADA.SampleFraction.Flag column for tracking.

File: R/ResultFlagsDependent.R [hunk 4]
```diff
@@ -412,17 +412,17 @@ InvalidResultUnit <- function(.data, clean = c("invalid_only", "nonstandardized_
...
   #   )
   
-  # if all rows are "Valid", return input unchanged
+  # if all rows are "Valid", return input with flag column
   if (any(c("Nonstandardized", "Invalid") %in%
           unique(check.data$TADA.ResultUnit.Flag)) == FALSE) {
-    print("All data are valid, therefore the function cannot be applied.")
-    .data = OrderTADACols(.data)
-    return(.data)
+    print("All characteristic/unit combinations are valid in your dataframe. Returning input dataframe with TADA.ResultUnit.Flag column.")
```
Comment (cristinamullin) [inline R/ResultFlagsDependent.R:418]: Suggestion: Returning input dataframe with added TADA.ResultUnit.Flag column for tracking.

File: R/Transformations.R [hunk 5]
```diff
@@ -103,8 +103,8 @@ ConvertResultUnits <- function(.data, transform = TRUE) {
     
     # Calculate deg F and deg C, replace Conversion factor values
     flag.data <- flag.data %>%
-      # apply function row by row
-      dplyr::rowwise() %>%
+      # apply function row by row - EDH - I don't think this is needed (I think default behavior of case_when is row by row)?
```
Comment (cristinamullin) [inline R/Transformations.R:106]: Agree, okay to fully remove where used

**Review by cristinamullin — APPROVED** [review]

File: R/ResultFlagsDependent.R (hunk 2 above)
Comment (ehinman) [reply R/ResultFlagsDependent.R:250]: I will double check but I'm pretty sure check.data in these functions is simply the input dataframe with the flag column added. Do you want me to rename check.data to just .data?

**Pull request merged**